src/function/utils/isPrime.js

```diff
@@ -35,39 +35,37 @@ export const createIsPrime = /* #__PURE__ */ factory(name, dependencies, ({ type
    */
   const isPrime = typed(name, {
     number: function (x) {
-      if (x < 2) {
+      if (x * 0 !== 0) {
         return false
       }
-      if (x === 2) {
-        return true
+      if (x <= 3) {
+        return n > 1
       }
-      if (x % 2 === 0) {
+      if (x % 2 === 0 || x % 3 == 0) {
         return false
       }
-      for (let i = 3; i * i <= x; i += 2) {
-        if (x % i === 0) {
+      for (let i = 5; i * i <= x; i += 6) {
+        if (x % i === 0 || x % (i + 2) == 0) {
           return false
         }
       }
       return true
     },
 
-    BigNumber: function (x) {
-      if (x.lt(2)) {
-        return false
-      }
-      if (x.equals(2)) {
-        return true
-      }
-      if (x.mod(2).isZero()) {
-        return false
+    BigNumber: function (n) {
+      if (n.toNumber() * 0 !== 0) {
```

Comment:
PeterOlson's lib has `isPrime` for BigIntegers - https://github.com/peterolson/BigInteger.js/blob/master/BigInteger.js#L917 
I believe, it is based on https://en.wikipedia.org/wiki/Miller–Rabin_primality_test#Deterministic_variants .

Reply:
Yeah, but if you read the Wikipedia article, you discover that's it's not 100% accurate. A such algorithm is definitely fast, but as we can't be certain if a given n is a prime or not, I would strongly suggest not to use it.

Reply:
@arguiot , are you sure? I could be wrong, but I think, what Wikipedia article says is that the algorithm is based on something unproved, but it is 100% accurate